docs/design/2021-03-09-security-enhanced-mode.md

```diff
@@ -0,0 +1,225 @@
+# Proposal:
+
+- Author(s):     [morgo](https://github.com/morgo)
+- Last updated:  March 9, 2021
+- Discussion at: N/A
+
+## Table of Contents
+
+* [Introduction](#introduction)
+* [Motivation or Background](#motivation-or-background)
+* [Detailed Design](#detailed-design)
+* [Test Design](#test-design)
+    * [Functional Tests](#functional-tests)
+    * [Scenario Tests](#scenario-tests)
+    * [Compatibility Tests](#compatibility-tests)
+    * [Benchmark Tests](#benchmark-tests)
+* [Impacts & Risks](#impacts--risks)
+* [Investigation & Alternatives](#investigation--alternatives)
+* [Unresolved Questions](#unresolved-questions)
+
+## Introduction
+
+This document was created to discuss the design of Security Enhanced Mode. It comes from the DBaaS requirement that `SUPER` users must not be able to perform certain actions that could comprimise the system.
+
+### Terminology
+
+* **Configuration Option**: The name of a variable as set in a configuration file.
+* **System Variable** (aka sysvar): The name of a variable that is set in a running TiDB server using the MySQL protocol.
+* **Super**: The primary MySQL "admin" privilege, which is intended to be superseded by MySQL’s "dynamic" (fine-grained) privileges starting from MySQL 8.0.
+
+## Motivation or Background
+
+Currently the MySQL `SUPER` privilege encapsulates a very large set of system capabilities. It does not follow the best practices of allocating _fine grained access_ to users based only on their system-access requirements.
+
+This is particularly problematic in a DBaaS scenario such as TiDB Cloud where the `SUPER` privilege has elements that are required by both end users (TiDB Cloud Customers) and system operations (PingCAP SREs).
+
+The design of Security Enhanced Mode (SEM) takes the approach of:
+
+1. Restricting `SUPER` to a set of capabilities that are safe for end users.
+2. Implementation of dynamic privileges ([issue #22439](https://github.com/pingcap/tidb/issues/22439)).
+
+This approach was requested by product management based on the broad "in the wild" association of `SUPER` as "the MySQL admin privilege". Thus, proposals to create a new lesser-`SUPER` privilege have already been discussed and rejected.
+
+The design and name of "Security Enhanced" is inspired by prior art with SELinux and AppArmor.
+
+## Detailed Design
+
+A boolean option called `EnableEnhancedSecurity` (default `FALSE`) will be added as a TiDB configuration option.  The following subheadings describe the behavior when `EnableEnhancedSecurity` is set to `TRUE`.
+
+### System Variables
+
+The following system variables will be hidden:
+
+* variable.TiDBDDLSlowOprThreshold,
```

Comment:
How about `variable.TiDBRedactLog`? @morgo @SunRunAway

Reply:
Yes - I agree this should be here, but I will defer to @SunRunAway to confirm.

Reply:
Agree.